src/_tutorials/dart-vm/httpserver.md

```diff
@@ -656,66 +649,66 @@ join() method in Stream to concatenate the string values of those chunks.
 The `basic_writer_server.dart` file implements
 a server that follows this pattern.
 
+<?code-excerpt "httpserver/bin/basic_writer_server.dart" replace="/\/\*\d\*\/|contentType.*? == \S+|(jsonData|jsonString|filename) = .*?;/[!$&!]/g"?>
 {% prettify dart %}
-  import 'dart:io';
-  import 'dart:convert';
-
-  main() async {
-    var server =
-        await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4049);
-    await for (var req in server) {
-      ContentType contentType = req.headers.contentType;
-
-      if (req.method == 'POST' &&
-[[note]]1[[/note]]       [[highlight]]contentType != null &&[[/highlight]]
-          [[highlight]]contentType.mimeType == 'application/json'[[/highlight]]) {
-        try {
-[[note]]2[[/note]]       [[highlight]]var jsonString = await req.transform(UTF8.decoder).join();[[/highlight]]
-
-          // Write to a file, get the file name from the URI.
-[[note]]3[[/note]]       var filename = [[highlight]]req.uri.pathSegments.last;[[/highlight]]
-          await new File(filename).writeAsString(jsonString,
-              mode: FileMode.WRITE);
-[[note]]4[[/note]]       Map jsonData = [[highlight]]JSON.decode(jsonString);[[/highlight]]
-          req.response..statusCode = HttpStatus.OK
-                      ..write('Wrote data for ${jsonData['name']}.')
-                      ..close();
-        } catch (e) {
-          req.response..statusCode = HttpStatus.INTERNAL_SERVER_ERROR
-                      ..write("Exception during file I/O: $e.")
-                      ..close();
-        }
-      } else {
-        req.response..statusCode = HttpStatus.METHOD_NOT_ALLOWED
-                    ..write("Unsupported request: ${req.method}.")
-                    ..close();
+import 'dart:async';
+import 'dart:io';
+import 'dart:convert';
+
+Future main() async {
+  var server = await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4049);
```

Comment:
The last visible characters on this line are `V4,`.

Reply:
fixed